contracts/TopShot.cdc

```diff
@@ -654,10 +667,184 @@ pub contract TopShot: NonFungibleToken {
+                    let squareImage = MetadataViews.Media(
+                        file: MetadataViews.HTTPFile(
+                            url: "https://nbatopshot.com/static/img/og/og.png"
+                        ),
+                        mediaType: "image/png"
+                    )
+                    return MetadataViews.NFTCollectionDisplay(
+                        name: "NBA-Top-Shot",
+                        description: "NBA Top Shot is your chance to own, sell, and trade official digital collectibles of the NBA and WNBA's greatest plays and players",
+                        externalURL: MetadataViews.ExternalURL("https://nbatopshot.com"),
+                        squareImage: squareImage,
+                        bannerImage: bannerImage,
+                        socials: {
+                            "twitter": MetadataViews.ExternalURL("https://twitter.com/nbatopshot"),
+                            "discord": MetadataViews.ExternalURL("https://discord.com/invite/nbatopshot"),
+                            "instagram": MetadataViews.ExternalURL("https://www.instagram.com/nbatopshot")
+                        }
+                    )
+                case Type<MetadataViews.Traits>():
+                    // non play specific traits
+                    let traitDictionary: {String: AnyStruct} = {
+                        "Series Number": TopShot.getSetSeries(setID: self.data.setID),
+                        "Set Name": TopShot.getSetName(setID: self.data.setID),
+                        "Serial Number": self.data.serialNumber
+                    }
+                    // add play specific data
+                    let fullDictionary = self.mapPlayData(dict: traitDictionary)
+                    return MetadataViews.dictToTraits(dict: fullDictionary, excludedNames: [])
+                case Type<MetadataViews.Medias>():
+                    return MetadataViews.Medias(
+                        items: [
+                            MetadataViews.Media(
+                                file: MetadataViews.HTTPFile(
+                                    url: self.mediumimage()
+                                ),
+                                mediaType: "image/jpeg"
+                            ),
+                            MetadataViews.Media(
+                                file: MetadataViews.HTTPFile(
+                                    url: self.video()
+                                ),
+                                mediaType: "video/mp4"
+                            )
+                        ]
+                    )
             }
 
             return nil
         }   
+
+        // Functions used for computing MetadataViews 
+
+        // mapPlayData helps build our trait map from play metadata
+        // Returns: The trait map with all non-nil fields from play data added
+        pub fun mapPlayData(dict: {String: AnyStruct}) : {String: AnyStruct} {      
+            let playMapping: {String: String} = {
```

Comment:
do we have this just for the display name? It doesn't seem extensible because if we decide to add extra metadata to new plays we mint, we will have to edit this contract. I'm kinda okay with us showing CamelCase for the display name

Reply:
Yes, just for display name -> I was going to say we had this also had this map to exclude some fields from getting included as traits, but that can probably better be accomplished by using the 'excludedFields' parameter.
Have updated -> now we'l use the names already present in the metadata map. Double checking, it looks like camel case names are appear in examples of Trait view usage, so I suspect UpperCamelCase should also be okay.